Mathlib/GroupTheory/SchurZassenhaus.lean

```diff
@@ -61,7 +61,7 @@ theorem smul_diff_smul' [hH : Normal H] (g : Gᵐᵒᵖ) :
         simp only [Subtype.ext_iff, coe_mk, coe_mul, mul_assoc, mul_inv_cancel_left] }
   refine (Fintype.prod_equiv (MulAction.toPerm g).symm _ _ fun x ↦ ?_).trans (map_prod ϕ _ _).symm
   simp only [smul_apply_eq_smul_apply_inv_smul, smul_eq_mul_unop, mul_inv_rev, mul_assoc,
-    MonoidHom.id_apply, toPerm_symm_apply, MonoidHom.coe_mk, OneHom.coe_mk]
+    MonoidHom.id_apply, id_eq, toPerm_symm_apply, MonoidHom.coe_mk, OneHom.coe_mk]
```

Comment:
Could you explain where the `id_eq`s come from? (Also in other files.) This is not too much of a regression but might imply the simp set is suboptimal.

Reply:
There are a lot of `id x` which appear when applying `map_smul` (because of the new type of equivariant maps), and they need to be rewritten.

Reply:
Maybe, we should rw `map_smul` when there is no `id`, and `map_smul_sl` otherwise.

Reply:
No longer needed:
```suggestion
    MonoidHom.id_apply, toPerm_symm_apply, MonoidHom.coe_mk, OneHom.coe_mk]
```